docs/cache/pattern/api-caching.md

```diff
@@ -0,0 +1,165 @@
+
+1. [Create a cache in the Momento console](https://console.gomomento.com/caches/create)
+2. [Obtain an API Key (AppId) from WolframAlpha](https://developer.wolframalpha.com)
+3. To call WolframAlpha API to retrieve a string description of the current weather, you can create a client, `WeatherDescriptionClient`, that does the following:
+```typescript
+import {City, WeatherDescriptionClient} from './weather-description-client';
+
+// We list only selected cities for brevity.
+export enum City {
+  NewYork,
+  LosAngeles,
+  Chicago,
+  Houston,
+  Phoenix,
+}
+
+export interface WeatherDescriptionClient {
+  getWeatherDescription(city: City): Promise<string>;
+}
+
+export class WolframWeatherDescriptionClient
+  implements WeatherDescriptionClient
+{
+  secretAppId: string;
+  constructor() {
+    const envSecretAppId = process.env['WOLFRAM_APP_ID'];
+    if (!envSecretAppId) {
+      throw new Error('Must provide WOLFRAM_APP_ID env variable');
+    }
+    this.secretAppId = envSecretAppId;
+  }
+
+  async getWeatherDescription(city: City): Promise<string> {
+    const cityName = City[city];
+    const url = `https://api.wolframalpha.com/v1/result?appid=${this.secretAppId}&i=Weather for ${cityName}`;
+    const response = await fetch(url);
+    if (response.ok) {
+      return await response.text();
+    } else {
+      const errorString = await response.text();
+      throw new Error(`Wolfram returned an error: ${errorString}`);
+    }
+  }
+}
+```
+4. Since the weather is unlikely to change that frequently for a particular city, and since a call to a remote API can be expensive, we can cache the results using [Momento Javascript SDK](../../sdks).
+For code reusability, we use the Decorator Pattern to wrap `CachingWeatherDescriptionClient` around an abstract `WeatherDescriptionClient` that we have created in an earlier step. Note that this requires no code-change to the existing `WolframWeatherDescriptionClient`.
+```typescript
+import {CacheClient, CacheGet} from '@gomomento/sdk';
+
+// This is the name of the distributed Cache that we will use
+// to store weather results for different cities.
+// This Cache must be created beforehand.
+const CACHE_NAME = 'weather-descriptions';
+
+// Weather data is highly cacheable since it is not likely to change rapidly.
+// Here we can configure the TTL for how long we want to cache the weather data for.
+const WEATHER_CACHE_TTL_SECONDS = 300;
+
+// This is an inline distributed Cache.
+// Results are served from the Cache whenever possible.
+// On a Cache miss, the actual service is called, then
+// the result will be populated into the Cache automatically.
+export class CachingWeatherDescriptionClient
+  implements WeatherDescriptionClient
+{
+  delegate: WeatherDescriptionClient;
+  momentoCacheClient: CacheClient;
+  constructor(
+    delegate: WeatherDescriptionClient,
+    momentoCacheClient: CacheClient
+  ) {
+    this.delegate = delegate;
+    this.momentoCacheClient = momentoCacheClient;
```

Comment:
I recommend using this instead since we're pushing that as the best practice in JS/TS.
```
this.momentoCacheClient = momentoCacheClient.cache(CACHE_NAME);
```
Then all references to it no longer have to pass in the cache name.

Reply:
Full disclosure, i don't remember if `.cache` is an async call or not. We need to verify.